commons/src/main/java/io/aiven/kafka/tieredstorage/commons/transform/FetchChunkEnumeration.java

```diff
@@ -0,0 +1,147 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.commons.transform;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
+import org.apache.kafka.server.log.remote.storage.RemoteLogSegmentMetadata;
+
+import io.aiven.kafka.tieredstorage.commons.Chunk;
+import io.aiven.kafka.tieredstorage.commons.ChunkManager;
+import io.aiven.kafka.tieredstorage.commons.manifest.SegmentManifest;
+import io.aiven.kafka.tieredstorage.commons.manifest.index.ChunkIndex;
+
+import org.apache.commons.io.input.BoundedInputStream;
+
+public class FetchChunkEnumeration implements Enumeration<InputStream> {
+    private final ChunkManager chunkManager;
+    private final RemoteLogSegmentMetadata remoteLogSegmentMetadata;
+    private final SegmentManifest manifest;
+    private final int startPosition;
+    private final int endPosition;
+    final int startChunkId;
+    final int endChunkId;
+    private final ChunkIndex chunkIndex;
+    int currentChunkId;
+
+    /**
+     *
+     * @param chunkManager provides chunk input to fetch from
+     * @param remoteLogSegmentMetadata required by chunkManager
+     * @param manifest provides to index to build response from
+     * @param startPosition original offset range start position
+     * @param endPosition original offset range end position
+     */
+    public FetchChunkEnumeration(final ChunkManager chunkManager,
+                                 final RemoteLogSegmentMetadata remoteLogSegmentMetadata,
+                                 final SegmentManifest manifest,
+                                 final int startPosition,
+                                 final int endPosition) {
+        this.chunkManager = Objects.requireNonNull(chunkManager, "chunkManager cannot be null");
+        this.remoteLogSegmentMetadata =
+            Objects.requireNonNull(remoteLogSegmentMetadata, "remoteLogSegmentMetadata cannot be null");
+        this.manifest = Objects.requireNonNull(manifest, "manifest cannot be null");
+        this.startPosition = startPosition;
+        this.endPosition = endPosition;
+
+        this.chunkIndex = manifest.chunkIndex();
+
+        final Chunk firstChunk = getFirstChunk(startPosition);
+        startChunkId = firstChunk.id;
+        currentChunkId = startChunkId;
+        final Chunk lastChunk = getLastChunk(endPosition);
+        endChunkId = lastChunk.id;
```

Comment:
Can we use same naming everywhere? e.g. start-end vs first-last. IMO the latter is a little bit better.